### PR TITLE
[RDY] Mansion Upgrade Project

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
Added mansion mapgen from Mansion Upgrade Project (see http://smf.cataclysmdda.com/index.php?topic=13630.0).

Thanks to @egomassive (http://smf.cataclysmdda.com/index.php?action=profile;u=3495).

### Todo

- [x] - test loot;
- [x] - utilize mapgen palettes:
- [x] - fix open air tiles;
- [x] - fix vegetation tiles underground.

### CODE REVIEW

**Comment by AdonaiJr** [comment]
Yes! Thank you so much for pushing this, Zhilkin. Egomassive made an awesome job at those mansions. It's a pity that his mod was buried in the foruns, and was not working for a while.

**Comment by Coolthulhu** [comment]
Needs a rebase, badly

**Comment by ZhilkinSerg** [comment]
> Needs a rebase, badly

I've just did something wrong when trying to rebase and resolve conflicts.

Will try to rollback.

File: data/json/mapgen/mansion/mansion.json [hunk 1]
```diff
@@ -0,0 +1,5841 @@
...
+    "method": "json",
+    "om_terrain": [
+      "mansion_t",
+      "mansion_t7"
+    ],
+    "weight": 1000,
+    "object": {
+      "fill_ter": "t_floor",
+      "rows": [
+        "X....nnn.X....X...
```
Comment (Coolthulhu) [inline data/json/mapgen/mansion/mansion.json]: This looks really wrong: `chance: 100` means only 1 in 100 safes will have anything in it.

Comment (Coolthulhu) [inline data/json/mapgen/mansion/mansion.json]: Terrain definitions should be in a separate file.

Comment (ZhilkinSerg) [reply data/json/mapgen/mansion/mansion.json]: Is it? I thought 100 is for 100%.

**Comment by ZhilkinSerg** [comment]
Fixed eveything reviewуd, except for palettes and item spawn chances.

File: data/json/mapgen/mansion/mansion.json (hunk 1 above)
Comment (ZhilkinSerg) [reply data/json/mapgen/mansion/mansion.json]: There were no issues in amount of loot when this mansion was in the mod. Will test it again later.

Comment (Coolthulhu) [reply data/json/mapgen/mansion/mansion.json]: Oh, I re-read it now. It looks like there are two different functions in mapgen and the item group one uses `x_in_y( chance, 100 )`, while `spawn_item` for some reason uses `one_in( chance )`. This is pretty terrible, but looks like the PR is not affected.

**Comment by ZhilkinSerg** [comment]
I'm still working on utilizing palettes.

**Comment by ZhilkinSerg** [comment]
The loot seems to be okay.

**Comment by ZhilkinSerg** [comment]
The masterpiece should be ready now.

**Comment by Coolthulhu** [comment]
Got some weird rock-filled rooms here and there:
rock room

**Comment by Coolthulhu** [comment]
Pretty much every mansion has this weird room of rock. There is no door leading to it, so I guess it's minor, but it should be fixed.
I guess I'll let it in for now and just report it, since players are very unlikely to see it.

**Comment by ZhilkinSerg** [comment]
> Pretty much every mansion has this weird room of rock. There is no door leading to it, so I guess it's minor, but it should be fixed.
> I guess I'll let it in for now and just report it, since players are very unlikely to see it.

What floor is it? Was it basement?

**Pull request merged**